### PR TITLE
Build Krew archive

### PR DESCRIPTION
**Pull request by adrianludwin**
Unlike in 0.6+, this change doesn't attempt to build the manifest as a
part of Cloud Build. That can be done by hand after the release; too
much has changed in the makefiles between the two versions. So this just
focuses on creating the archive.

Tested: this is virtually impossible to test given the state of the
build process in v0.5 (it's much better in v0.6). I'll revert the
changes to cloudbuild.yaml if I can't build a release candidate, and the
changes to the hack/krew*.yaml files don't matter to anyone.

### CODE REVIEW

File: incubator/hnc/cloudbuild.yaml [hunk 1]
```diff
@@ -32,6 +32,11 @@ steps:
     # Build plugin
     GOOS=linux GOARCH=amd64 go build -o kubectl-hns_linux_amd64 ../cmd/kubectl/main.go
     GOOS=darwin GOARCH=amd64 go build -o kubectl-hns_darwin_amd64 ../cmd/kubectl/main.go
+    mkdir kubectl
+    cp kubectl-hns_linux_amd64 kubectl/
+    cp kubectl-hns_darwin_amd64 kubectl/
+    cp ../LICENSE kubectl/
+    tar czvf kubectl-hns.tar.gz kubectl/
```
Comment (yiqigao217) [inline incubator/hnc/cloudbuild.yaml:39]: Why didn't see this part in #1136 but here?

Comment (adrianludwin) [reply incubator/hnc/cloudbuild.yaml:39]: In v0.6, cloudbuild.yaml defers to the makefile for almost everything. In v0.5 it replicates what's in the makefile, so I had to do the same.

File: incubator/hnc/hack/krew-kubectl-hns.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
```
Comment (rjbez17) [inline incubator/hnc/hack/krew-kubectl-hns.yaml:1]: In the 0.6 PR you use sed to replace the tag, checksum, etc in this file. I don't see and of that here, am I missing something?

Comment (rjbez17) [reply incubator/hnc/hack/krew-kubectl-hns.yaml:1]: oh I see, you meant to do it by hand. Maybe add a comment indicating as such? Or purposely making it invalid yaml so it isn't pushed accidentally?

Comment (adrianludwin) [reply incubator/hnc/hack/krew-kubectl-hns.yaml:5]: Big scary comment added :)

**Pull request merged**